common/scala/src/main/scala/whisk/core/database/CouchDbRestStore.scala

```diff
@@ -52,7 +52,8 @@ class CouchDbRestStore[DocumentAbstraction <: DocumentSerializer](
     dbPort: Int,
     dbUsername: String,
     dbPassword: String,
-    dbName: String)(implicit system: ActorSystem, val logging: Logging, jsonFormat: RootJsonFormat[DocumentAbstraction])
+    dbName: String,
+    cleanup:(String)=>Unit)(implicit system: ActorSystem, val logging: Logging, jsonFormat: RootJsonFormat[DocumentAbstraction])
```

Comment:
Why do you need to pass the `dbName` in? When you open the client you automatically have it at hand.

`shutdown: => Unit` should suffice (note the whitespace please)